### ISSUE TITLE
[Bug]: Highland gov uk source doesn't work with my route but does with the example one

### ISSUE THREAD

**Issue by Supermillhouse** [mampfes/hacs_waste_collection_schedule#2196]
### I Have A Problem With:

A specific source

### What's Your Problem

I am unable to get any of the dates for my route to show up, I am using the highland_gov_uk source.

I have tested it successfully with one of the uprn that [5ila5](https://github.com/mampfes/hacs_waste_collection_schedule/commits?author=5ila5) used during debugging and everything works ok.

The configuration below is an address that is on my route, but not my address, it also doesn't work.
I am currently on build 72f99a9

### Source (if relevant)

highland_gov_uk

### Logs

```Shell
no relevant logs
```

### Relevant Configuration

```YAML
waste_collection_schedule:
    sources:
    - name: highland_gov_uk
      args:
        uprn: 130143631
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by 5ila5** [comment]
If you search on the website for the address associated with this uprn shows `We cannot find collection details for this address. Contact recycle@highland.gov.uk for assistance.`

You can check here: <https://highland-self.achieveservice.com/en/service/Check_your_household_bin_collection_days> 

You can see the UPRN associated with the address by inspecting the address drop down.

**Comment by Supermillhouse** [comment]
My apologies, this address is probably part of a commercial collection as it is a Jehovah Witnesses community building.

This is a house down the road, 130072429

I used [this](https://www.findmyaddress.co.uk/search) site to double check the address and the UPRN, this one should be good, I also tested it in Home Assistant and it also fails to list the collection details in the calendar.

**Comment by Supermillhouse** [comment]
I was just looking at the [rows_data][0] for my address and it looks like it only populates data into the keys that end with "old".
I believe your code uses the keys ending with new.
One of the addresses you used has data in both old and new, but on the gov website only displays the "new" data.

so... if "new" is null use "old" else use new?

**Comment by 5ila5** [comment]
Fixed in #2201

I thought don't use the Old values anymore, so I ignored them.
